### PR TITLE
Remove slicemachine toggle and related code

### PR DESCRIPTION
**Pull request by gestchild**
For #10781 

This turned into quite a large PR, but contains lots of deleted files and small changes to files, so hopefully isn't too onerous.

What it does:
- removes the sliceMachine toggle
- removes the code we used to use for rendering body slices (we now do this with Prismic's SliceZone component)
- removes the old body and standfirst props, we now use untransformedBody and untransformedStandfirst (this resulted in having to do a few code changes, as we were still using body in a few places even if the sliceMachine toggle was on)
- updates some of the cardigan stories with data for the untransformedBody
- removes/updates mock content for tests where necessary
- moves us to using the types that prismic slice machine generates, rather than the ones we've written which are now removed (there may be a few more places we can do this, but I think I got most of them)
- fixes errors that arose after shifting to the prismic generated types
- deletes all the old prismic model files we no longer use
- removes the featured text from the whats-on page. It won't work since slice machine migration and wasn’t being used anyway. It was a covid era thing.
- changes some import urls to use absolute paths
- makes a couple of tweaks so the sliceAnalysis and diffCustomTypes scripts still work
- Lastly, I discovered Textured backgrounds aren't working on events and event-series. The code started erroring due to my changes, but since it was already broken, I just commented it out and have opened up a separate [issue for this](https://github.com/wellcomecollection/wellcomecollection.org/issues/10919) rather than tackle it here

### CODE REVIEW

File: common/server-data/prismic.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export type ResultsLite = {
 export const defaultValue = {
   globalAlert: {
     data: {
-      isShown: null,
+      isShown: 'hide' as 'hide' | 'show',
```
Comment (rcantin-w) [inline common/server-data/prismic.ts:28]: Can you explain the `'hide' as 'hide'` for my own knowledge gaining? :D

**Review by rcantin-w — DISMISSED** [review]
Is there a value in keeping `Prismic` in the name of types (e.g. `CollectionVenuePrismicDocument` becoming `CollectionVenueDocument`) to differentiate them from transformed content? (I think that's why we imported Prismic functions with `*` so we could read them as `prismic.function` )

It'd be a lot of renaming but just wanted to ask as we'd discussed the above example in the past.

File: content/webapp/pages/index.tsx [hunk 2]
```diff
@@ -12,7 +12,6 @@ import { ArticleBasic } from '@weco/content/types/articles';
 import Space from '@weco/common/views/components/styled/Space';
 import Layout, { gridSize10 } from '@weco/common/views/components/Layout';
 import SimpleCardGrid from '@weco/content/components/SimpleCardGrid/SimpleCardGrid';
-import PageHeaderStandfirst from '@weco/common/views/components/PageHeaderStandfirst/PageHeaderStandfirst';
```
Comment (rcantin-w) [inline content/webapp/pages/index.tsx:15]: Is it used anywhere else now? (should it be deleted)

File: content/webapp/services/prismic/transformers/body.ts [hunk 3]
```diff
@@ -439,21 +424,25 @@ export function transformContentListSlice(
         .map(content => {
           switch (content.type) {
             case 'pages':
-              return transformPage(content);
+              return transformPage(content as unknown as PagesDocument);
```
Comment (rcantin-w) [inline content/webapp/services/prismic/transformers/body.ts:427]: This one too I'd love to understand better, why is this needed?

File: content/webapp/services/prismic/types/images.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ type Params = {
 };
 
 export function getImageUrlAtSize(
-  image: Image | Picture,
+  image: Image | Omit<Image, '32:15' | '16:9' | 'square'> | Picture,
```
Comment (rcantin-w) [inline content/webapp/services/prismic/types/images.ts:12]: Might be worth adding a comment as to why we're omitting these crops?

File: content/webapp/types/body.ts [hunk 5]
```diff
@@ -43,29 +49,23 @@ export type Slice<TypeName extends string, Value> = {
...
 }
 
-export function isStandfirst(
-  slice: BodySlice
-): slice is BodySlice & { type: 'standfirst' } {
-  return slice.type === 'standfirst';
-}
+// export function isStandfirst(slice: prismic.Slice): slice is StandfirstSlice {
+//   return slice.slice_type === 'standfirst';
+// } // TODO is this used?
```
Comment (rcantin-w) [inline content/webapp/types/body.ts:68]: Should this be done/removed?

File: content/webapp/services/prismic/transformers/body.ts [hunk 6]
```diff
@@ -75,6 +73,7 @@ import { Props as TitledTextListProps } from '@weco/content/components/TitledTex
 import { Props as AsyncSearchResultsProps } from '@weco/content/components/SearchResults/AsyncSearchResults';
 import { Venue } from '@weco/common/model/opening-hours';
 
+// TODO all the TODOs in this file
```
Comment (rcantin-w) [inline content/webapp/services/prismic/transformers/body.ts:76]: 😄 what's this/should it be addressed as part of this?

Comment (gestchild) [reply content/webapp/services/prismic/transformers/body.ts:76]: oh yeah, I think I've the ones I cared about, so I'll remove this comment. It was just a reminder to self. There's 2 TODOs that David added that concern labels, but they can be done separately and another one that predates the slice machine work, which again can be addressed separately.

File: content/webapp/services/prismic/types/images.ts (hunk 4 above)
Comment (gestchild) [reply content/webapp/services/prismic/types/images.ts:12]: I actually don't think we need to do this, I'll remove it. Think I put it in temporarily as it was being noisy during the process of doing this work. Thanks for catching

File: content/webapp/pages/index.tsx (hunk 2 above)
Comment (gestchild) [reply content/webapp/pages/index.tsx:15]: it is used in the Standfirst slice file (might be worth renaming it at some point)

**Comment by gestchild** [comment]
> Is there a value in keeping `Prismic` in the name of types (e.g. `CollectionVenuePrismicDocument` becoming `CollectionVenueDocument`) to differentiate them from transformed content? (I think that's why we imported Prismic functions with `*` so we could read them as `prismic.function` )

Yes, it probably does - I'd just defaulted to what Prismic calls them, but I'll rename then when I import them. Thanks

File: common/server-data/prismic.ts (hunk 1 above)
Comment (gestchild) [reply common/server-data/prismic.ts:28]: If I give it the value 'hide' without the `as` there is a typescript error saying`Type 'string' is not assignable to type'"hide" | "show"'. So I had to tell it the value is one of those specific strings. It think `as const` would be better actually, then we wouldn't have to change this if we added more possible strings

File: content/webapp/services/prismic/transformers/body.ts (hunk 3 above)
Comment (gestchild) [reply content/webapp/services/prismic/transformers/body.ts:426]: The content can be one of many types. Each of the transform functions expects a particular one of those. Because of the switch statement we know what type we are dealing with so I tell typescript that is so it doesn't complain. Because the different possible types don't overlap sufficiently typescript complains unless it is converted to unknown first.

Another way to do it would be to write some type guards to identify each specific specific type.

**Comment by gestchild** [comment]
"Is there a value in keeping Prismic in the name of types"

I'm halfway through doing this, but wondering if it should be something like UntransformedCollectionVenueDocument or RawCollectionVenueDocument - in case we ever move away from Prismic

**Comment by rcantin-w** [comment]
> I'm halfway through doing this, but wondering if it should be something like UntransformedCollectionVenueDocument or RawCollectionVenueDocument - in case we ever move away from Prismic

I'm liking `Raw` in lieu of `Untransformed` anyway, clear _and_ shorter!
I wondered if the fact that `Document` was in the name was clarifying that point a little too, but possibly not, since it could be transformed... Naming things eh.

File: content/webapp/services/prismic/transformers/pages.ts [hunk 7]
```diff
@@ -12,14 +14,12 @@ import { transformSeason } from './seasons';
...
 import { Link } from '../../../types/link';
-import { Body } from '../types/body';
-import { SeasonPrismicDocument } from '../types/seasons';
 import { transformContributors } from './contributors';
 import { isNotUndefined, isUndefined } from '@weco/common/utils/type-guards';
 import { transformTimestamp } from '@weco/common/services/prismic/transformers';
 import { SiteSection } from '@weco/common/views/components/PageLayout/PageLayout';
 
-export function transformOnThisPage(body: Body): Link[] {
+export function transformOnThisPage(body): Link[] {
```
Comment (rcantin-w) [inline content/webapp/services/prismic/transformers/pages.ts:22]: Were there issues with this type?
